### ISSUE TITLE
webapp:What extension is required to run the Front Door Access Restriction example on a webapp?

### ISSUE THREAD

**Issue by plamber** [Azure/azure-cli#17931]
Hello,
I was trying to run the example provided on the page:

```
az webapp config access-restriction add -g ResourceGroup -n AppName --priority 400 --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=12345678-abcd-1234-abcd-12345678910a
```

I am receiving the error message *unrecognized arguments: --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=12345678-abcd-1234-abcd-12345678910a*

My CLI was upgraded to the latest version. I assume that I need to install a preview extension to make this work but I am not able to find it.

Can somebody help me out?

Thank you,
Patrick
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c788e908-cee3-ae46-fac4-956d0606ad76
* Version Independent ID: 2ed69bfd-bb05-3d09-072e-00e505856fba
* Content: [az webapp config access-restriction](https://docs.microsoft.com/en-us/cli/azure/webapp/config/access-restriction?view=azure-cli-latest)
* Content Source: [latest/docs-ref-autogen/webapp/config/access-restriction.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/latest/docs-ref-autogen/webapp/config/access-restriction.yml)
* Service: **app-service-web**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yungezz** [comment]
route to appropriate team

**Comment by Kotasudhakarreddy** [comment]
@plamber - I am unable to repro the issue. Please try with latest CLI version `2.22.0`. If you still face the issue share us the `--debug` logs. PFB FYR.

![image](https://user-images.githubusercontent.com/60102891/116682610-1541fa00-a9cc-11eb-8590-a6f083fd0fe9.png)

**Comment by plamber** [comment]
Hi,
this is the version:

![image](https://user-images.githubusercontent.com/18241305/116686706-7e1d7800-a9b4-11eb-8779-973fd8ac968e.png)

```
cli.knack.cli: Command arguments: ['webapp', 'config', 'access-restriction', 'add', '-n', 'el-dev-api-eu-01-we', '--priority', '200', '--service-tag', 'AzureFrontDoor.Backend', '--http-header', 'x-azure-fdid=de0f2823-b95f-4697-94f7-4a4aae3a1354', '--debug']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x03245E88>, <function OutputProducer.on_global_arguments at 0x033E13D8>, <function CLIQuery.on_global_arguments at 0x033FB300>]
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'webapp': ['azure.cli.command_modules.appservice', 'azext_webapp']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands
cli.azure.cli.core: appservice                0.227        62       220
cli.azure.cli.core: Total (1)                 0.227        62       220
cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
cli.azure.cli.core: Loading extensions:
cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
cli.azure.cli.core: webapp                    0.003         4         8  C:\Users\PatrickLamber\.azure\cliextensions\webapp
cli.azure.cli.core: Total (1)                 0.003         4         8  
cli.azure.cli.core: Loaded 65 groups, 227 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : webapp config access-restriction add
cli.azure.cli.core: Command table: webapp config access-restriction add
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x036D8D68>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 'C:\Users\PatrickLamber\.azure\commands\2021-04-30.13-04-04.webapp_config_access-restriction_add.41376.log'.
az_command_data_logger: command args: webapp config access-restriction add -n {} --priority {} --service-tag {} --http-header {} --debug
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x03724468>, <function register_global_query_examples_argument.<locals>.register_query_examples at 0x03724BB8>]
cli.azure.cli.core.util: attempting to read file C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\command_modules\appservice\resources\WindowsFunctionsStacks.json as utf-8-sig
cli.azure.cli.core.util: attempting to read file C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\command_modules\appservice\resources\LinuxFunctionsStacks.json as utf-8-sig
cli.knack.commands: Configured default 'rg-el-dev-eu-01' for arg resource_group_name
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x03724C00>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x03724C90>]        
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.azure.cli.core.command_recommender: "--service-tag" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.command_recommender: "--http-header" is an invalid parameter for command "webapp config access-restriction add".
urllib3.connectionpool: Starting new HTTPS connection (1): app.aladdin.microsoft.com:443
urllib3.connectionpool: https://app.aladdin.microsoft.com:443 "GET /api/v1.0/suggestions?query=%7B%22command%22%3A+%22webapp+config+access-restriction+add%22%2C+%22parameters%22%3A+%22%22%7D&clientType=AzureCli&context=%7B%22versionNumber%22%3A+%222.22.1%22%2C+%22errorType%22%3A+%22UnrecognizedArguments%22%2C+%22correlationId%22%3A+%225a6e6439-51ff-4ec8-b0ba-2c448886e138%22%2C+%22subscriptionId%22%3A+%22da80e753-e223-4526-835c-6cc1ec387035%22%7D 
HTTP/1.1" 200 None
cli.azure.cli.core.command_recommender: "--service-tag" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.command_recommender: "--http-header" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.azclierror: unrecognized arguments: --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=de0f2823-b95f-4697-94f7-4a4aae3a1354
az_command_data_logger: unrecognized arguments: --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=de0f2823-b95f-4697-94f7-4a4aae3a1354
```

It is an Azure Linux Web Instance. Not sure if that matters.

br,
Patrick

**Comment by Kotasudhakarreddy** [comment]
@plamber It doesn't matter if it is windows or Linux web instance. It should work with any OS type. And I found there is no `-g` passed along with your command. 
I am unable to see the version you are using as image is not displaying. Please retry with the latest CLI version `2.22.0` if not already using.

**Comment by plamber** [comment]
Hi,
I am using the 2.22.1. I am not passing the -g parameter because I was setting it before using this command: az configure --defaults group=.... 

Even with the explicitly assigned -g command I am experiencing the same issue.

br,
Patrick

**Comment by Kotasudhakarreddy** [comment]
Hi , 

got you on how you are passing `-g`.  Could you try the same in Clod Shell (Azure portal) once.

**Comment by plamber** [comment]
Hi,
this is also failing.

Here the results from the Cloudshell.

```
az webapp config access-restriction add -g rg-el-dev-eu-01 -n el-dev-api-eu-01-we --priority 400 --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=12345678-abcd-1234-abcd-12345678910a --debug
cli.knack.cli: Command arguments: ['webapp', 'config', 'access-restriction', 'add', '-g', 'rg-el-dev-eu-01', '-n', 'el-dev-api-eu-01-we', '--priority', '400', '--service-tag', 'AzureFrontDoor.Backend', '--http-header', 'x-azure-fdid=12345678-abcd-1234-abcd-12345678910a', '--debug']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f9400a5ec80>, <function OutputProducer.on_global_arguments at 0x7f940098f7b8>, <function CLIQuery.on_global_arguments at 0x7f94009ab8c8>]
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'webapp': ['azure.cli.command_modules.appservice']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands
cli.azure.cli.core: appservice                0.181        62       220
cli.azure.cli.core: Total (1)                 0.181        62       220
cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_next']
cli.azure.cli.core: Loading extensions:
cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
cli.azure.cli.core: ai-examples               0.002         1         1  /opt/az/lib/python3.6/site-packages/azure-cli-extensions/ai-examples
cli.azure.cli.core: Total (1)                 0.002         1         1
cli.azure.cli.core: Loaded 63 groups, 221 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : webapp config access-restriction add
cli.azure.cli.core: Command table: webapp config access-restriction add
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f94004e6950>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '/home/patrick/.azure/commands/2021-05-04.10-29-38.webapp_config_access-restriction_add.199.log'.
az_command_data_logger: command args: webapp config access-restriction add -g {} -n {} --priority {} --service-tag {} --http-header {} --debug
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f940048bae8>, <function register_global_query_examples_argument.<locals>.register_query_examples at 0x7f940045fae8>]
cli.azure.cli.core.util: attempting to read file /opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/resources/WindowsFunctionsStacks.json as utf-8-sig
cli.azure.cli.core.util: attempting to read file /opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/resources/LinuxFunctionsStacks.json as utf-8-sig
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f940045fb70>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7f940045fc80>]
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.azure.cli.core.command_recommender: "--service-tag" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.command_recommender: "--http-header" is an invalid parameter for command "webapp config access-restriction add".
urllib3.connectionpool: Starting new HTTPS connection (1): app.aladdin.microsoft.com:443
urllib3.connectionpool: https://app.aladdin.microsoft.com:443 "GET /api/v1.0/suggestions?query=%7B%22command%22%3A+%22webapp+config+access-restriction+add%22%2C+%22parameters%22%3A+%22%22%7D&clientType=AzureCli&context=%7B%22versionNumber%22%3A+%222.22.0%22%2C+%22errorType%22%3A+%22UnrecognizedArguments%22%2C+%22correlationId%22%3A+%229da9a75c-adb4-4e67-a948-d6b12337dd8d%22%2C+%22subscriptionId%22%3A+%22da80e753-e223-4526-835c-6cc1ec387035%22%7D HTTP/1.1" 200 None
cli.azure.cli.core.command_recommender: "--service-tag" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.command_recommender: "--http-header" is an invalid parameter for command "webapp config access-restriction add".
cli.azure.cli.core.azclierror: unrecognized arguments: --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=12345678-abcd-1234-abcd-12345678910a
az_command_data_logger: unrecognized arguments: --service-tag AzureFrontDoor.Backend --http-header x-azure-fdid=12345678-abcd-1234-abcd-12345678910a

TRY THIS:
az webapp config access-restriction add --resource-group ResourceGroup --name AppName --rule-name developers --action Allow --ip-address 130.220.0.0/27 --priority 200
Add Access Restriction opening (Allow) named developers for IPv4 address 130.220.0.0/27 with priority 200 to main site.

az webapp config access-restriction add --resource-group ResourceGroup --name AppName --rule-name build_server --action Allow --ip-address 192.168.0.0/27 --priority 250 --scm-site true
Add Access Restriction opening (Allow) named build_server for IPv4 address 192.168.0.0/27 with priority 250 to scm site.

https://docs.microsoft.com/en-US/cli/azure/webapp/config/access-restriction#az_webapp_config_access_restriction_add
Read more about the command in reference docs
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x7f94004e6b70>]
az_command_data_logger: exit code: 2
cli.__main__: Command ran in 1.543 seconds (init: 0.149, invoke: 1.394)
telemetry.save: Save telemetry record of length 3605 in cache
telemetry.check: Negative: The /home/patrick/.azure/telemetry.txt was modified at 2021-05-04 10:28:37.218523, which in less than 600.000000 s
```

**Comment by Kotasudhakarreddy** [comment]
Hi can you please run `az webapp config access-restriction add -h` and check do you see `--service-tag` and ` --http-headers` arguments like below. If you don't see them please try re-install `CLI` and try again.

![image](https://user-images.githubusercontent.com/60102891/117138803-25414b80-adc9-11eb-80ae-930d465676ae.png)

**Comment by plamber** [comment]
Hi,
I did following:
- Reinstalled the CLI: no change
- Checked on the Cloudshell: no change
- Installed the beta version: no change

Is there anything else that must be added (an addon or something)?

br,
Patrick

**Comment by Kotasudhakarreddy** [comment]
Would you able to see `--service-tag` and `--http-headers` arguments when you run `az webapp config access-restriction add -h` from your CLI console.

**Comment by plamber** [comment]
No, in all cases the parameters were missing

**Comment by Kotasudhakarreddy** [comment]
@plamber This feature is implemented and released in CLI version 2.23.0. You can find the release notes [here](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/docs-ref-conceptual/release-notes-azure-cli.md#app-service). So please install the latest one try for the same.

**Comment by plamber** [comment]
Hi @Kotasudhakarreddy,
thank you. Was able to upgrade to 2.23.0 and was able to find the new properties.

br,
Patrick
